src/main/java/mate/academy/security/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.security.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.security.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()) {
+            throw new AuthenticationException("Email " + email + " don't registered");
+        }
+        User user = userFromDbOptional.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashedPassword)) {
+            return user;
+        } else {
+            throw new AuthenticationException("Wrong password");
+        }
```

Comment:
Invert if statement to reduce code
```suggestion
        if (!user.getPassword().equals(hashedPassword)) {
        throw new AuthenticationException("Wrong password");
        } 
        return user;
```